xls/dslx/stdlib/std.x

```diff
@@ -177,6 +177,53 @@ fn iterative_div_test () {
   ()
 }
 
+// Binary Division algorithm as outlined by:
+// http://www.dragonwins.com/domains/GetTechEd/de248/binary_division.htm
+fn iterative_binary_div<N: u32, DN: u32 = {N * u32:2}>(dividend: bits[N], divisor: bits[N]) -> (bits[N], bits[N]) {
+  let quotient: bits[N] = bits[N]:0;
+  let remainder: bits[N] = dividend;
+  let term: bits[DN] = bits[DN]:1 << (N as bits[DN]);
+  let product: bits[DN] = (divisor as bits[DN]) << (N as bits[DN]);
+  let (quotient, remainder, product, term) =
+     for (i, (quotient, remainder, product, term)): (bits[N], (bits[N], bits[N], bits[DN], bits[DN]))
```

Comment:
@grebe commented in #935 

> Making your loop indices u32 seems like it will make this a little more readable- I forget, can you do range(0, N) or will you have to do range(u32:0, N)